grpcutil/cancel.go

```diff
@@ -9,15 +9,14 @@ import (
 	"errors"
 
 	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
 )
 
 // IsCanceled checks whether an error comes from an operation being canceled
 func IsCanceled(err error) bool {
 	if errors.Is(err, context.Canceled) {
 		return true
 	}
-	s, ok := status.FromError(err)
+	s, ok := ErrorToStatus(err)
```

Comment:
Doing full conversion to gogo's `status.Status` seems unnecessary here, if all we care about is code from grpc.Status. Perhaps we can have `ErrorToStatusCode` function too?

Reply:
It is true that we need only the status code here, but in order to get it, we anyway need to call a `gogostatus.FromError()` or `grpcstatus.FromError()` (the current implementation).
My idea here was to unify the way we extract statuses. Because in most of the places it is done by `gogostatus`, and in a couple of places it is done by `grpcstatus`.

The idea is, anyway, to get rid of the `gogo`-related things, but once we find a valid alternative. If we now start mixing `gogo` and `grpc` usages, it will be more difficult to do the change one day.

Reply:
> If we now start mixing gogo and grpc usages, it will be more difficult to do the change one day.

I don't see how my proposal would do that?

Reply:
> I don't see how my proposal would do that?
I was referring to the current implementation: it added a `grpcstatus` usage to `dskit`. And my motivation with the proposed change was to get rid of it. I meant that.

Reply:
> I was referring to the current implementation: it added a `grpcstatus` usage to `dskit`. And my motivation with the proposed change was to get rid of it. I meant that.

We need to check for `GRPCStatus() *grpcstatus.Status` method, is this what you're referring to? I don't see a way around it.